mathics/builtin/datentime.py

```diff
@@ -1055,7 +1055,13 @@ def eval(self, n, evaluation):
             )
             return
 
-        time.sleep(sleeptime)
+        steps = int(1000 * sleeptime)
+        while steps > 0:
```

Comment:
`for _ in range(steps)` is a little nicer. 

In general, whenever I see a polling loop like this, I often wonder if there isn't a better way. I can't think of one offhand. 

When I look at https://reference.wolfram.com/language/ref/Pause.html I note a discrepancy in our description. 

We have

we have 

> pauses for _n_ seconds.


where WMA says:

> pauses for _at least n_ seconds. 

which, in fact, matches the behavior we have better.

Note also that by polling we do increase the `Timing` value a little better. And that's also another reason why using `time.sleep()` by itself would better.